modules/internal/ChapelArray.chpl

```diff
@@ -2142,13 +2145,15 @@ module ChapelArray {
   record _array {
     var _pid:int;  // only used when privatized
     pragma "owned"
+    pragma "alias scope from this"
     var _instance; // generic, but an instance of a subclass of BaseArr
     var _unowned:bool;
 
     proc chpl__promotionType() type {
       return _value.eltType;
     }
 
+    pragma "alias scope from this"
```

Comment:
pragmas like this also make me nervous because they are bound to be forgotten about when new methods are added

Reply:
This pragma isn't strictly necessary but I wanted the no-alias analysis to be able to determine when the result of a dsiAccess call has the same "scope" as the array it comes from.

In the DefaultRectangular example I started with (tuplesVSarrays), this pragma isn't necessary because dsiAccess is inlined.

This pragma is a performance hint.